go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -387,14 +390,31 @@ func (qre *QueryExecutor) checkPermissions() error {
 		remoteAddr = ci.RemoteAddr()
 		username = ci.Username()
 	}
-	action, desc := qre.plan.Rules.GetAction(remoteAddr, username, qre.bindVars, qre.marginComments)
+
+	bufferingTimeoutCtx, cancel := context.WithTimeout(qre.ctx, maxQueryBufferDuration)
```

Comment:
this is apretty cool PR, i had a question - the fresh 10s timer applies on a per query basis. So if for some reason cutover is taking long - we will see 10s latency before the precondition failure. Is this intentional or should this instead be a shared context across all queries which will cause fail fast on preconditions for any cutover taking longer than 10s?

Reply:
I'm not sure I understand the question. If the cutover is taking long we will see up to 10sec latency -- that's the expected behavior. A query must wait for up to 10sec. However, it does not wait the entire 10s if the cutover is either complete or failed. When that happens the wait is terminated prematurely.

Reply:
Thanks i got the answer, but to add more details of my question

Scenario:
* Cutover took 60 seconds: Say it started at T0 and finished at T0+60s
* queries come in at the rate of 1 per second - ie. 1 query at T0, 1 at T0 + 1s ... 1 at T0+59s and so on

After this change:
* every query between T0 and T0+50s is rejected after 10 seconds latency
* queries between T0 + 51s and T0 + 59s will suceed due to the upto 10s buffer

My question was, would it instead be better to have a static buffer of 10s since cutover start ie. 
* any query between T0 and T0+10s gets buffered and rejected after 10s in this scenario
* queries between T0+11s and T0+59s get rejected immediately as buffer period of 10s since start of schema migration has expired
* queries coming in after T0+60s succeed as cutover has completed and write-block is cleared

I now understand that it is the former (just wanted to understand the behavior better)

Reply:
Got it, thanks for clarifying! I'm open to alternatives if there's a clear advantage to one over the other, such as preserving resources. If you think a specific approach is better than another in all/most cases, then let's try and look at the numbers/advantage over a new issue!